packages/shared/src/components/modals/LockedSquadModal.tsx

```diff
@@ -38,15 +41,23 @@ function LockedSquadModal({
   onRequestClose,
 }: LockedSquadModalProps): ReactElement {
   const { showPrompt } = usePrompt();
+  const client = useQueryClient();
+  const { displayToast } = useToastNotification();
   const [copying, copyLink] = useCopyLink(() => squad.permalink);
   const onCopy = () => {
     copyLink();
   };
 
   const onDeleteSquad = async (e) => {
-    if (await showPrompt(options)) {
+    if (!(await showPrompt(options))) {
+      return;
+    }
+    try {
       await deleteSquad(squad.id);
+      await client.invalidateQueries(BOOT_QUERY_KEY);
```

Comment:
Not sure if manipulating the boot query directly is the best idea.
Alternatively we can manually set different data, or refetch the list of active squads to push over it?

Reply:
@idoshamun / @sshanzel Would love to hear how you think about invalidating it.

Reply:
IIRC invalidating the cache would trigger `refetch`, which I think should be fine, but the concern would be is right after the async call of `deleteSquad`, the `boot` data from `gateway` would already be updated (literally split second after the call). I think locally modifying here can also be an option, but if the current procedure works, then I'm cool with it.

Reply:
If there's a way to prevent the unnecessary call, it would be better I think. I agree we can simply patch the data ourselves.

Reply:
I have added a new utility function to fetch and update only the squads